lib/Deployer.js

```diff
@@ -20,12 +20,13 @@ class Deployer {
      *
      * @param {TYPE.Mcdevrc} properties General configuration to be used in retrieve
      * @param {TYPE.BuObject} buObject properties for auth
+     * @param {boolean} [fromRetrieve] optionally deploy whats defined via selectedTypesArr + keyArr directly from retrieve folder instead of from deploy folder
      */
-    constructor(properties, buObject) {
+    constructor(properties, buObject, fromRetrieve) {
         this.buObject = buObject;
         this.properties = properties;
         this.deployDir = File.normalizePath([
-            properties.directories.deploy,
+            fromRetrieve ? properties.directories.retrieve : properties.directories.deploy,
```

Comment:
why not just change "properties" value in the one location? This would be strongly preferred from my side, since it shouldnt really matter what the deploy folder is

Reply:
that will only work after your PR was successfully merged. at the time of writing this code, your PR still had a lot of todos and uncertainties. 

if we manage to get concerns taken care of in your PR AND all automated & manual additional tests to work... which is a lot, then we could change that

Reply:
My PR wouldnt affect it - this would just mean overriding the properties values before passing into deployer. Also, is there anything stopping someone from just changing the config file?